src/wymeditor/editor/base.js

```diff
@@ -1803,6 +1803,30 @@ WYMeditor.editor.prototype.insertImage = function (attrs) {
 };
 
 /**
+    editor._updateImageAttrs
+    ========================
+
+    Updates provided `img`'s attributes with provided `attrs`.
+*/
+WYMeditor.editor.prototype._updateImageAttrs = function (img, attrs) {
+    var $img = jQuery(img);
+    if (attrs.src !== $img.attr('src')) {
+        // this data is used in the ImageHandler.
+        // height/width ratio is now most likely invalid
+        $img.data('DimensionsRatio', null);
+
+        // since the height/width ration is most likely different
+        // these proportions are most likely wrong.
+        // easiest solution is to remove them
+        // and let the user deal with
+        // the real size of the source image
+        // and he'll be able to resize it
+        $img.removeAttr('height').removeAttr('width');
```

Comment:
this might lead us in circles a bit, but I wonder if it might be worth keeping the height/width, if the new dimensions are the same?  this might lead to usability improvements in some cases (for example, if you are swapping out a logo or diagram and want it to take up the same amount of space)